### PR TITLE
Just use a formatter...

### PR DESCRIPTION
**Pull request by chiroptical**
Use a `fourmolu.yaml` file to format all of the Haskell. Use `alejandra` to format the nix.

Check both of these in CI before doing anything else.

### CODE REVIEW

**Comment by chessai** [comment]
Rebase and this looks good (according to the formatter, at least)

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,11 @@ ghcid-test: hpack
 	ghcid -c cabal repl test:spec
 
 format:
-	find src/ test/ -name "*.hs" -exec fourmolu -i {} --ghc-opt -XTypeApplications \;
+	alejandra --quiet .
```
Comment (chiroptical) [inline Makefile:23]: If we are going to format, we may as well to nix too.

File: Makefile [hunk 2]
```diff
@@ -20,7 +20,11 @@ ghcid-test: hpack
...
+	fourmolu -i src/ test/
+
+check-format:
+	fourmolu -m check src/ test/
```
Comment (chiroptical) [inline Makefile]: I want this in CI

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -18,4 +18,5 @@ jobs:
           extra-experimental-features = nix-command flakes ca-derivations
           extra-substituters = https://cache.nixos.org/ https://cache.garnix.io/
           extra-trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g=
+    - run: nix develop --command make check-format
```
Comment (chiroptical) [inline .github/workflows/build-and-test.yml:21]: Don't even try to `nix build` if the format is wrong.

File: .github/workflows/nix.yml [hunk 4]
```diff
@@ -32,5 +32,6 @@ jobs:
           require-sigs = true
           sandbox = true
           extra-system-features = nixos-test benchmark big-parallel kvm
+    - run: nix develop --command make check-format
```
Comment (chiroptical) [inline .github/workflows/nix.yml:35]: Don't even try to `nix {develop,build}` if the format is wrong.

**Review by chessai — APPROVED** [review]

**Pull request merged**